src/translate.rs

```diff
@@ -1855,6 +1937,120 @@ impl FromGlibPtrContainer<*const c_char, *mut glib_sys::GHashTable> for HashMap<
     }
 }
 
+impl<T> FromGlibContainerAsVec<<T as GlibPtrDefault>::GlibType, *mut glib_sys::GPtrArray> for T
+where
+    T: GlibPtrDefault
+        + FromGlibPtrNone<<T as GlibPtrDefault>::GlibType>
+        + FromGlibPtrFull<<T as GlibPtrDefault>::GlibType>,
+{
+    unsafe fn from_glib_none_num_as_vec(ptr: *mut glib_sys::GPtrArray, num: usize) -> Vec<T> {
+        if num == 0 || ptr.is_null() {
+            return Vec::new();
+        }
+        let pdata = (*ptr).pdata;
+        assert!((*ptr).len as usize >= num);
+        let mut res = Vec::with_capacity(num);
+        for i in 0..num {
+            let item_ptr: <T as GlibPtrDefault>::GlibType = Ptr::from(ptr::read(pdata.add(i)));
+            if !item_ptr.is_null() {
+                res.push(from_glib_none(item_ptr));
+            }
+        }
+        res
+    }
+
+    unsafe fn from_glib_container_num_as_vec(ptr: *mut glib_sys::GPtrArray, num: usize) -> Vec<T> {
+        let res = FromGlibContainer::from_glib_none_num(ptr, num);
+        if !ptr.is_null() {
+            glib_sys::g_ptr_array_free(ptr, false.to_glib());
+        }
+        res
+    }
+
+    unsafe fn from_glib_full_num_as_vec(ptr: *mut glib_sys::GPtrArray, num: usize) -> Vec<T> {
+        if num == 0 || ptr.is_null() {
+            return Vec::new();
+        }
+        let pdata = (*ptr).pdata;
+        assert!((*ptr).len as usize >= num);
+        let mut res = Vec::with_capacity(num);
+        for i in 0..num {
+            let item_ptr: <T as GlibPtrDefault>::GlibType = Ptr::from(ptr::read(pdata.add(i)));
+            if !item_ptr.is_null() {
+                res.push(from_glib_full(item_ptr));
```

Comment:
This is a bit problematic as we free the array, and it might then free its items with the free function configured on it.

I think for the time being we have to use `from_glib_none()` here until we have high-level bindings for the pointer array instead of converting to a vec.

Reply:
This is now done as well.